Source/Diagnostics/ComputeDiagFunctors/RhoFunctor.cpp

```diff
@@ -19,17 +23,35 @@ RhoFunctor::operator() ( amrex::MultiFab& mf_dst, const int dcomp, const int /*i
     auto& warpx = WarpX::GetInstance();
     std::unique_ptr<amrex::MultiFab> rho;
 
+    // Deposit charge density
+    // Call this with local=true since the parallel transfers will be handled
+    // by ApplyFilterandSumBoundaryRho
+
     // Dump total rho
     if (m_species_index == -1) {
         auto& mypc = warpx.GetPartContainer();
-        rho = mypc.GetChargeDensity(m_lev);
+        rho = mypc.GetChargeDensity(m_lev, true);
     }
     // Dump rho per species
     else {
         auto& mypc = warpx.GetPartContainer().GetParticleContainer(m_species_index);
-        rho = mypc.GetChargeDensity(m_lev);
+        rho = mypc.GetChargeDensity(m_lev, true);
     }
 
+    // Call this in case filtering is turned on
```

Comment:
About this comment, I think `ApplyFilterandSumBoundaryRho` does something (sum over guard cells) even when filtering is turned off. I'm afraid that this comment might be misleading (the call has an effect only in case filtering is turned on), but maybe I misunderstood something.

Reply:
Yes, I agree. I'll fix it to make it more clear what is happening.

Previously, the parallel sum over guard cells was being done inside the call to GetChargeDensity. Now, the ApplyFilterandSumBoundaryRho is used instead since it will handle both the filtering and the parallel sum. The comment means that ApplyFilterandSumBoundaryRho is used instead of SumBoundary (as called in GetChargeDensity) since it would do the filtering if needed.